test/input_gen/recorder.py

```diff
@@ -390,3 +389,41 @@ def record(
             train_step(model, optimizer, loss_fn, initial_input, label, write, **kwargs)
 
         # self.inference_step(initial_input, label, write)
+
+
+##
+# @brief record a single layer
+def record_single(layer, input_shape, file_name):
+    layer = attach_trans_layer(layer)
+    inputs = _rand_like(input_shape)
+
+    with tf.GradientTape(persistent=True) as tape:
+        tape.watch(inputs)
+        outputs = layer(inputs)
+        dy_constant = outputs * 2 # set incoming derivative to 2 instead of 1
```

Comment:
how about dy_constant = _rand_like(output_shape)?

Reply:
This adds additional overhead of setting into run context, saving derivatives and stuff.
While it is a good point, I'll leave it as a todo for now but work within next PR :)

Reply:
cool.